tests/functional/generate_pages.py

```diff
@@ -647,11 +645,61 @@ def process_schema(in_schema, out_dir, spec_file, require_path=".."):
         os.mkdir(out_dir)
 
     for section in data["sections"]:
+        if "summary" in section:
+            process_section_summary(
+                section["id"], out_dir, data, spec_file, require_path
+            )
         for group in section["groups"]:
             for block in group["blocks"]:
                 process_block(block, out_dir, data, spec_file, require_path)
 
 
+def process_section_summary(
+    section_id,
+    dir_out,
+    schema_data,
+    spec_file,
+    relative_require="..",
+    page_filename=None,
+):
+
+    logger.debug("Processing section summary: %s", section_id)
+
+    if not page_filename:
+        page_filename = f"{section_id}-summary.page.js"
+
+    page_path = os.path.join(dir_out, page_filename)
+
+    logger.info("creating %s...", page_path)
+
+    with open(page_path, "w") as page_spec:
+
+        section_context = {
+            "pageName": generate_pascal_case_from_id(section_id),
+            "basePage": "QuestionPage",
+            "basePageFile": "question.page",
+            "pageDir": dir_out.split("/")[-1],
+            "pageFile": page_filename,
+            "type_id": section_id,
+            "type_name": camel_case(generate_pascal_case_from_id(section_id)),
+            "relativeRequirePath": relative_require,
+        }
+
+        page_spec.write(HEADER.substitute(section_context))
+        page_spec.write(CLASS_NAME.substitute(section_context))
+        page_spec.write(CONSTRUCTOR.substitute(section_context))
+        process_summary(schema_data, page_spec, False, section_summary=True)
```

Comment:
Calling `process_summary` here adds methods for all answers in the questionnaire to every section summary page file e.g. `generated_pages/section_summary/house-details-section-summary.page.js` and `generated_pages/section_summary/property-details-section-summary.page.js` are almost identical.

Reply:
Happy to fix, but looking at the code it looks like its always been bloated
```
if block["type"] in ("Summary", "SectionSummary"):
            collapsible = block.get("collapsible", False)
            process_summary(schema_data, page_spec, collapsible)
```
Summary and SectionSummary had no separation in what was passed, therefore it would always have been a bit on the chubby side

Reply:
Ah, fair point. I will raise a card for fixing that separately.

Reply:
No need, already done it, the whole py needs looking at. It really could do with some rationalising and tiding up. In another places I fixed a for loop that was followed by the same loop of things, when it only need to go through once.

Reply:
Yeah, it's a bit of a mess. I will amend the card i raised to look into refactoring it 😀.